reservation/interfaces.go

```diff
@@ -0,0 +1,121 @@
+package reservation
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/loop/fsm"
+	reservation_script "github.com/lightninglabs/loop/reservation/script"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+var (
+	ErrReservationAlreadyExists = fmt.Errorf("reservation already exists")
+	ErrReservationNotFound      = fmt.Errorf("reservation not found")
+)
+
+const (
+	// TODO(sputn1ck): decide on actual value.
+	KeyFamily         = int32(42068)
+	DefaultConfTarget = int32(3)
+)
+
+// Store is the interface that stores the reservations.
+type Store interface {
+	// StoreReservation stores the reservation in the database.
+	CreateReservation(ctx context.Context, reservation *Reservation) error
+
+	// UpdateReservation updates the reservation in the database.
+	UpdateReservation(ctx context.Context, reservation *Reservation) error
+
+	// GetReservation retrieves the reservation from the database.
+	GetReservation(ctx context.Context,
+		id ID) (*Reservation, error)
+
+	// ListReservations lists all existing reservations the client has ever
+	// made.
+	ListReservations(ctx context.Context) ([]*Reservation, error)
+}
+
+// ID is a unique identifier for a reservation.
+type ID [32]byte
+
+// FromByteSlice creates a reservation id from a byte slice.
+func (r *ID) FromByteSlice(b []byte) error {
+	if len(b) != 32 {
+		return fmt.Errorf("reservation id must be 32 bytes, got %d, %x",
+			len(b), b)
+	}
+
+	copy(r[:], b)
+	return nil
+}
+
+// Reservation holds all the necessary information for the 2-of-2 multisig
+// reservation utxo.
+type Reservation struct {
+	ID ID
+
+	State fsm.StateType
+
+	ClientPubkey *btcec.PublicKey
+	ServerPubkey *btcec.PublicKey
+	Value        btcutil.Amount
+	Expiry       uint32
+
+	KeyLocator keychain.KeyLocator
+
+	Outpoint         *wire.OutPoint
+	InitiationHeight int32
+
+	ConfirmationHeight uint32
+}
+
+func (r *Reservation) GetPkScript() ([]byte, error) {
+	if r.ClientPubkey == nil {
+		return nil, errors.New("client pubkey is nil")
+	}
+
+	if r.ServerPubkey == nil {
```

Comment:
Alternatively could add a `NewReservation` function that sets these fields and does the prechecks.